### DISCUSSION TITLE
Using Raft cluster in an IIS application

### DISCUSSION THREAD

**Discussion by simontherry** [dotnet/dotNext#115] [Q&A]
When using the RAFT HTTP implementation, adding a node is done through cluster.AddMemberAsync, which requires a HTTP endpoint object as input. How does it work when you are running multiple webservices in the same IIS where some of them use this RAFT implementation? Does each of the applications require a different port to be opened? Can they share the 80/443 port?

**Comment by sakno** [comment]
@simontherry , the simple answer is to use different domain names. Using the same IIS for hosting multiple nodes violates the key aspect of Raft - nodes must be hosted independently (at least in containers). For instance, if your host with IIS crashed, multiple nodes become unavailable which impacts the entire cluster and the majority. If nodes are running on the same machine, you can use `localhost` address space with different ports.

P.S.: The issue converted to discussion.

**Reply by sakno** [reply]
@simontherry , good news for you. In upcoming 4.7.0 release, I'm moving from `HttpEndPoint` to [UriEndPoint](https://docs.microsoft.com/en-us/dotnet/api/microsoft.aspnetcore.connections.uriendpoint) as a part of [Microsoft.AspNetCore.Connections](https://docs.microsoft.com/en-us/dotnet/api/microsoft.aspnetcore.connections) adaptation for Raft library. Abstractions provided by this library allow me completely split network transport from serialization and other Raft-specific stuff.

`UriEndPoint` will allow to distinguish cluster nodes based on their URI. It means that you can host multiple nodes on the same DNS and even on the same port. However, I do not recommend to do that because you're losing fault tolerance.
